src/app/BreedProcessor.ts

```diff
@@ -50,35 +46,38 @@ export class BreedProcessor {
   }
 
   processBreedPhase({ theHerd }: Player): RollResult {
-    const wolfDiceResult = this.randomResultInterfaceWolf.getRandomValue();
-    const foxDiceResult = this.randomResultInterfaceFox.getRandomValue();
-    const equalResult = foxDiceResult === wolfDiceResult;
-    const roll = [wolfDiceResult, foxDiceResult];
-    if (equalResult) {
-      const count = this.breedAnimals(foxDiceResult, theHerd, true);
-      return { rollResult: roll, gain: [[foxDiceResult, count]] };
+    const roll = [
+      this.firstDice.getRandomValue(),
+      this.secondDice.getRandomValue(),
+    ];
+    const [firstDice, secondDice] = roll;
+    if (firstDice === secondDice) {
+      const count = this.breedAnimals(firstDice, theHerd, true);
+      return {
+        rollResult: roll,
+        gain: count ? [[firstDice, count]] : [],
+      };
     }
     const gain: [AnimalNames, number][] = [];
-    if (foxDiceResult === AnimalNames.FOX) {
-      const fox = this.getPredatorByName(AnimalNames.FOX);
-      this.returnToBank(fox, theHerd);
-      theHerd.cullAnimals(fox, this.mode);
-    } else {
-      gain.push([
-        foxDiceResult,
-        this.breedAnimals(foxDiceResult, theHerd),
-      ]);
-    }
-    if (wolfDiceResult === AnimalNames.WOLF) {
-      const wolf = this.getPredatorByName(AnimalNames.WOLF);
-      this.returnToBank(wolf, theHerd);
-      theHerd.cullAnimals(wolf, this.mode);
-    } else {
-      gain.push([
-        wolfDiceResult,
-        this.breedAnimals(wolfDiceResult, theHerd),
-      ]);
-    }
+    roll
+      .filter((animal) => !this.isPredator(animal))
+      .forEach((animal) => {
+        const grow = this.breedAnimals(animal, theHerd);
+        if (grow) {
+          gain.push([animal, grow]);
+        }
+      });
+    roll
+      .filter((animal) => this.isPredator(animal))
+      .forEach((animal) => {
+        const predator = this.getPredatorByName(animal);
+        const isHerdCulled = this.returnToBank(predator, theHerd);
+        if (isHerdCulled && gain.length > 0) {
+          this.reduceGain(predator, gain);
+        }
+        predator.attackHerd();
+        theHerd.cullAnimals(predator, this.mode);
+      });
```

Comment:
In 62 there is a check if none of the animals is a predator, then in 71 is a check if it is a predator. Perhaps it could be checked just once and work on a base of if/else statements ?(maybe with early return statement so no else would be needed?)

Reply:
I prefer to work on arrays and streams, it's faster and removes many shifts for if